llvm/lib/Target/X86/GISel/X86InstructionSelector.cpp

```diff
@@ -195,6 +195,14 @@ X86InstructionSelector::getRegClass(LLT Ty, const RegisterBank &RB) const {
       return &X86::VR512RegClass;
   }
 
+  if (RB.getID() == X86::PSRRegBankID) {
+    if (Ty.getSizeInBits() == 80)
+      return &X86::RFP80RegClass;
+    if (Ty.getSizeInBits() == 64)
+      return &X86::RFP64RegClass;
+    return &X86::RFP32RegClass;
```

Comment:
Sorry, I don't understand this comment. Could you please rephrase?

Reply:
This is about whether it is guaranteed that type sizes of `PSRRegBankID` are limited by `32`, `64` and `80`. Or do we have to check on `32` as well to hit `llvm_unreachable` if something out of these three types comes.

I'm also thinking about whether it is better to rewrite this function to use switches instead of if/else...

Reply:
If I query for X86::PSRRegBankID and 31 bits, I would get X86::RFP32RegClass. This is bad. See the other groups. They check for all legal sizes.